src/astra_driver.cpp

```diff
@@ -199,6 +203,18 @@ bool AstraDriver::EnableStreaming(std_srvs::SetBool::Request &req, std_srvs::Set
   return true;
 }
 
+void AstraDriver::setHealthTimers() {
+  auto reset_this = [this](const ros::TimerEvent&) -> void
+  {
+    ROS_WARN_STREAM("Astra " << ns_ << " driver timeout! Resetting");
+    auto& nh = nh_;
+    auto& pnh = pnh_;
+    this->~AstraDriver();
+    new (this) AstraDriver(nh, pnh);
+  };
+  depth_callback_timer_ = nh_.createTimer(depth_callback_timeout_, reset_this, false, false);
```

Comment:
`auto reset_this = [this](const ros::TimerEvent&) -> void `
Can reset_this be void?

Reply:
the lambda return type is void